lib/auth.coffee

```diff
@@ -251,10 +272,12 @@ getAuth = (deps, opts) ->
 	# });
 	###
 	exports.getUserId = (callback) ->
-		token.getUserId().then (id) ->
-			throw new errors.ResinNotLoggedIn() if not id?
-			return id
-		.asCallback(callback)
+		exports.getUserDetails()
+			.then (body) ->
+				return body.id
+			.catch ->
+				throw new errors.ResinNotLoggedIn()
```

Comment:
Reminder: If the `getUserDetails()` method starts returning a `ResinNotLoggedIn` error, then this can be removed and you could end up with a method as simple as `exports.getUserDetails().get('id')`.